js/playlists.view.js

```diff
@@ -0,0 +1,165 @@
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License,
+// version 3, along with this program.  If not, see
+// http://www.gnu.org/licenses/
+//
+
+/*! Zookeeper Online (C) 1997-2020 Jim Mason <jmason@ibinx.com> | @source: https://zookeeper.ibinx.com/ | @license: magnet:?xt=urn:btih:1f739d935676111cfff4b4693e3816e664797050&dn=gpl-3.0.txt GPL-v3.0 */
+
+$().ready(function(){
+    // fetch the days with playlists for the selected month and enable the on the calendar.
+    function setAvailableDays(showDate) {
+        let url = '/index.php?action=playlistDaysByDate&viewdate=' + showDate;
+        $.ajax({
+            dataType : 'json',
+            type: 'GET',
+            accept: "application/json; charset=utf-8",
+            url: url,
+        }).done(function (goodDays) {
+            let excludeDays = $('td[data-handler="selectDay"]').filter(function(){
+                let excludeDay = !goodDays.includes($(this).text());
+                return excludeDay;
+            });
+            excludeDays.unbind('click').addClass(['ui-datepicker-unselectable', 'ui-state-disabled']);
+        });
+    }
+
+    // get playlists for a given ISO date & update the display table.
+    function getPlaylists(isoDate) {
+        let timestamp = Date.parse(isoDate + 'T00:00:00');
+        if (!timestamp)
+            errMsg = 'Date entry ' + dateStr + ' is not a valid date.';
+        else if (playlistStartTimestamp && timestamp < playlistStartTimestamp) {
+            alert("Sorry, the earliest playlist date is " + playlistStartIso);
+            return;
+        }
+        
+        setDateDisplay(isoDate);
+        let url = '/index.php?action=playlistsByDate&viewdate=' + isoDate;
+        $('#playlist-list tbody').html('');
+        $.ajax({
+            dataType : 'json',
+            type: 'GET',
+            accept: "application/json; charset=utf-8",
+            url: url,
+        }).done(function (respObj) {
+            let options = { weekday: 'long', year: 'numeric', month: 'short', day: 'numeric' };
+            let date = new Date(isoDate  + 'T00:00:00');
+            prettyDate = date.toLocaleString('default', options);
+            $('#playlist-list thead th').html(`Playlists for ${prettyDate}:`);
+            $('#playlist-list tbody').html(respObj.tbody);
+        });
+    }
+
+    function setDateDisplay(isoDate) {
+        let dateAr = isoDate.split('-');
+        let localDate = `${dateAr[1]}-${dateAr[2]}-${dateAr[0]}`;
+        if (isUSLocale) 
+            localDate = `${dateAr[1]}/${dateAr[2]}/${dateAr[0]}`;
+            
+        $("#playlist-date").val(localDate);
+    }
+
+    let locale = (new Intl.NumberFormat()).resolvedOptions().locale
+    let isUSLocale = locale.endsWith('-US');
+    $("#playlist-date").attr('placeholder', (isUSLocale ? 'mm/dd/yyyy<ENTER>' : 'dd-mm-yyyy<ENTER>'));
```

Comment:
Can we show the calendar on entry to the page?

For keyboard users, they can just key in a date and enter, but mouse-driven users are forced to do an extra click, just to get to the calendar.

As a compromise, on entry to the page, I suggest pre-opening the calendar but then setting focus into the input field.  That way, keyboard users can just start typing away, while users driving with the mouse have immediate access to the calendar without an extra click.

Once a date is picked, I think it's fine to close the calendar and force the user to click the calendar button if he wants to see it again.  It's only the case of entry to the page that the calendar should show by default.

Reply:
I'm not so sure about this one. Standard behavior for DPs is to display the calendar upon user activation and this is how the date & time pickers in the playlist edit page work. Also, since the page is entered via a click the user will already be in mouse mode so clicking the calendar is not much additional work.

Reply:
I'm prepared to take it without this change, but I do feel it would be a better UX if the calendar were displayed on entry.

Reply:
I feel that this is inconsistent with standard DP operation. Suggest that we go without this and see if anyone has any opinions on it. On a somewhat related issue I pushed a change to populate the current day's lists on page entry

Reply:
I like what you've done with it now, defaulting to the current day.  Thanks also for the fixes.